app/product/deleteProduct.js

```diff
@@ -0,0 +1,50 @@
+const Joi = require('joi');
+const { validator } = require('../util/validator');
+const { sendError } = require('../util/error.js');
+const { ObjectId } = require('mongodb');
+
+exports.deleteProduct_v1 = (req, res, next) => {
+    validateDeleteProductRequest(req)
+    .then((payload) => deleteProduct(payload))
+    .then(() => { res.json({ 
+        code: 200, 
+    }) }).catch((errorCode) => {
+        switch(errorCode) {
+            case 401: return sendError(res, errorCode, 'Product not found!');
+            case 422: return sendError(res, errorCode, 'Invalid request!');
+            case 504: return sendError(res, errorCode, 'Unable to delete product from store!');
+            default: return sendError(res, 500, 'Something went wrong!');
+        }
+    })
+}
+
+function validateDeleteProductRequest(request) {
+    const payload = request.query;
+    return new Promise(function (resolve, reject) {
+        const schema = Joi.object({
+            productId: Joi.string().trim().required()
+        });
+        const { error, value } = validator(schema, payload);
+        
+        if(error) {
+            console.log(error.details);
+            reject(422);
+        } 
+        resolve(value);
+    });
+}
+
+function deleteProduct(payload) {
+    return new Promise(function (resolve, reject) {
+        return require('../database.js').getDb().collection('product').deleteOne({"_id": new ObjectId(payload.productId)})
+        .then((response) => {
+            if(response.deletedCount === 0)
+                reject(401);
+            resolve();
```

Comment:
- I can change this to deleteCount === 1 but since mongo guarantees unique keys so delete count will either be 1 or 0 but never be more than 1. 
- Logging will be out of scope for now so once all services are done then will create Logger factory and logs everything.

Reply:
Good comment.

Reply:
I see you are using `deleteOne`. i was worried that more than 1 would be deleted but looks good as is.